util/dvsim/SimCfg.py

```diff
@@ -507,13 +526,18 @@ def _create_deploy_objects(self):
                 test.build_mode = Modes.find_mode(
                     build_map[test.build_mode].name, self.build_modes)
 
-        if self.run_only:
-            self.builds = []
-            build_map = None
-
         self.runs = ([]
                      if self.build_only else self._expand_run_list(build_map))
 
+        # Discard the build_job dependency that was added earlier if --run-only
+        # switch is passed. This is a little cumbersome than previous approach,
+        # but needed so that we can accurately set the index and total for each
+        # run job.
```

Comment:
It's not great to talk about the history of the file in a comment, because a reader then needs to go and understand what the file used to look like. Probably it would be better to say something like "We can't just throw away the build_job dependency before calling `_expand_run_list` because ...".